src/app/SharpRaven/Data/Context/Device.cs

```diff
@@ -0,0 +1,220 @@
+        /// </summary>
+        [JsonProperty(PropertyName = "external_free_storage", NullValueHandling = NullValueHandling.Ignore)]
+        public long? ExternalFreeStorage { get; set; }
+        /// <summary>
+        /// A formatted UTC timestamp when the system was booted.
+        /// </summary>
+        /// <example>
+        /// 018-02-08T12:52:12Z
+        /// </example>
+        [JsonProperty(PropertyName = "boot_time", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTimeOffset? BootTime { get; set; }
+        /// <summary>
+        /// The timezone of the device.
+        /// </summary>
+        /// <example>
+        /// Europe/Vienna
+        /// </example>
+        [JsonIgnore]
+        public TimeZoneInfo Timezone { get; set; }
+
+        /// <summary>
+        /// Clones this instance
+        /// </summary>
+        /// <returns></returns>
+        internal Device Clone()
+        {
+            return new Device
+            {
+                Name = this.Name,
+                Architecture = this.Architecture,
+                BatteryLevel = this.BatteryLevel,
+                BootTime = this.BootTime,
+                ExternalFreeStorage = this.ExternalFreeStorage,
+                ExternalStorageSize = this.ExternalStorageSize,
+                Family = this.Family,
+                FreeMemory = this.FreeMemory,
+                FreeStorage = this.FreeStorage,
+                MemorySize = this.MemorySize,
+                Model = this.Model,
+                ModelId = this.ModelId,
+                Orientation = this.Orientation,
+                Simulator = this.Simulator,
+                StorageSize = this.StorageSize,
+                Timezone = this.Timezone,
+                UsableMemory = this.UsableMemory
+            };
+        }
+
+        /// <summary>
+        /// Creates a Device while attempting to detect the running device info.
+        /// </summary>
+        /// <returns>An instance if any property was successfuly set or null otherwise.</returns>
+        internal static Device Create()
+        {
+            try
+            {
+                var device = new Device();
+
+                device.Name = Environment.MachineName;
+                device.Timezone = TimeZoneInfo.Local;
+                device.Architecture = GetArchitecture();
+                device.BootTime = new DateTimeOffset(
+                    DateTime.UtcNow - TimeSpan.FromTicks(Stopwatch.GetTimestamp()),
+                    TimeSpan.Zero);
+
+                return device;
+            }
+            catch (Exception e)
+            {
+                SystemUtil.WriteError(e);
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the architecture of this device.
+        /// </summary>
+        /// <returns>Arch or null if unknown</returns>
+        internal static string GetArchitecture() =>
```

Comment:
All these `internal` methods only seem to be used in a `private` scope, so can they be changed to `private` instead?

Reply:
Yes, I made them internal for testing but I'm just covering them through testing the whole Device at the end. I'll make them private.

Reply:
It is actually used by tests*